_source/_posts/2022-04-21-jhipster-k8s-azure.md

```diff
@@ -0,0 +1,669 @@
+
+The [JHipster API Gateway](https://www.jhipster.tech/api-gateway/) is the public face of your microservice. All public traffic comes through this service, which also includes the Vue frontend. The gateway is one of three application types that can be created by the JHipster generator DSL. The other two are monolith and microservice. A monolith is a non-microservice application with a single service.
+
+The store service and blog service are both examples of the microservice application type. This means that they are a Spring Boot resources server with some type of SQL or NoSQL backend.
+
+The generator creates four applications. They are designed to be built and run as docker containers, which makes it easy for them to be packaged in [Kubernetes](https://kubernetes.io/docs/concepts/overview/what-is-kubernetes/) pods. Kubernetes is a container orchestrator specifically designed for managing microservice networks. It's something like Docker Compose but designed for microservices with a lot of great features like service discovery, load balancing, automatic rollouts and restarts, resource management, and storage mounting.
+
+#### Prerequisites
+
+This tutorial has a lot of pieces. Install the required software below and sign up for an Azure Cloud account. You'll need a free Okta account, but you can use the Okta CLI to sign up for it later in the tutorial.
+
+- [Docker](https://docs.docker.com/get-docker/): you'll need to have both **Docker Engine** and **Docker Compose** installed (If you install the docker desktop, this will automatically install both. On Linux, if you install Docker Engine individually, you will have to also [install Docker Compose](https://docs.docker.com/compose/install/)) separately.
+- [Docker Hub](https://hub.docker.com/): you'll need this to host the docker images so that Azure can pull them
+- [Java 11](https://adoptopenjdk.net/)+
+- [Okta CLI](https://cli.okta.com/manual/#installation): you'll use Okta to add security to the microservice network. You can register for a free account from the CLI.
+- [Azure Cloud account](https://azure.microsoft.com/en-us/free/): they offer an account with a $200 credit to start out. 
+- [Azure CLI](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli): you'll use the Azure CLI to manage the Kubernetes cluster
+- [kubectl](https://kubernetes.io/docs/tasks/tools/): CLI to manage Kubernetes clusters
+
+#### Modifying the Generated JHipster Microservice Project for Azure and Cosmos DB
+
+This project is based on two of Matt Raible's tutorials: *[Reactive Java Microservices with Spring Boot and JHipster](https://developer.okta.com/blog/2021/01/20/reactive-java-microservices)* and *[Kubernetes to the Cloud with Spring Boot and JHipster](https://developer.okta.com/blog/2021/06/01/kubernetes-spring-boot-jhipster)*. In these tutorials, he builds a reactive Java microservice and shows how to deploy it to Google Cloud (GCP). I have modified the project to work with Azure and Cosmos DB.
```

Comment:
I'm not sure these blog posts should be italicized, but maybe. I'll leave this here for @havi to decide.